File: python/algorithms/Primality Testing.py
# Program for checking whether given number is a prime or not(Primality Testing) in O(sqrt(n)).

import math

def isPrime(n):
	if n < 2:
		return False
	if n == 2:
		return True
	if(n % 2 == 0):
		return false
	for i in range(3, int(math.sqrt(n)), 2):
		if n % i == 0:
			return False
	return True

def main():
	print("Enter any number: ")
	n = int(input())
	if isPrime(n):
		print("It is a prime number.")
	else:
		print("It's not a prime number!")

if __name__ == '__main__':
	main()

File: cpp/algorithms/Primality Testing.cpp
// Program for checking whether given number is a prime or not(Primality Testing) in O(sqrt(n)).

#include <iostream>
#include <cmath>

using namespace std;

bool isPrime(int n)
{
	if(n < 2)	return false;
	for(int i=2; i<=sqrt(n); i++)
	{
		if(n % i == 0)
			return false;
	}
	return true;
}

int main()
{
	int n;
	cout << "Enter any number: ";
	cin >> n;
	if(isPrime(n))	cout << "It is a prime number." << endl;
	else	cout << "It's not a prime number!" << endl;
	return 0;
}
